src/svg-icon.jsx

```diff
@@ -112,12 +105,17 @@ const SvgIcon = React.createClass({
       ...other,
     } = this.props;
 
+    const {
+      baseTheme,
+      prepareStyles,
+    } = this.state.muiTheme;
+
     const offColor = color ? color :
       style && style.fill ? style.fill :
-      this.state.muiTheme.rawTheme.palette.textColor;
+      baseTheme.palette.textColor;
     const onColor = hoverColor ? hoverColor : offColor;
 
-    const mergedStyles = this.mergeStyles({
+    const mergedStyles = Object.assign({
```

Comment:
Yeah this goes the same thing we discussed in the other issue! I'd like to remove this too if it's fine :grin: 

Let me know if you think it's okay.